packages/embeds/js/src/features/blocks/inputs/number/components/NumberInput.tsx

```diff
@@ -14,24 +16,22 @@ type NumberInputProps = {
 };
 
 export const NumberInput = (props: NumberInputProps) => {
-  const [inputValue, setInputValue] = createSignal<string | number>(
-    props.defaultValue ?? "",
-  );
-  const [staticValue, bindValue, targetValue] = numberInputHelper(() =>
-    inputValue(),
-  );
+  const numberInput = useNumberInput({
+    formatOptions: {
+      style: props.block.options?.style,
+      currency: props.block.options?.currency,
+      unit: props.block.options?.unit,
+    },
+  });
   let inputRef: HTMLInputElement | undefined;
 
-  const checkIfInputIsValid = () =>
-    inputRef?.value !== "" && inputRef?.reportValidity();
-
   const submit = () => {
-    if (checkIfInputIsValid())
+    if (!numberInput().invalid) {
```

Comment:
We used `reportValidity` to have at least the Browser system message displayed like this:

![CleanShot 2025-03-18 at 16 06 09@2x](https://github.com/user-attachments/assets/9d8858da-0874-4c4b-a95d-9acaa661b2a4)

Is there a way to still have that even when using thie Ark component?

Reply:
Still, it doesn't work because ArkUI under the hood is not providing the native HTML attributes to the number input. Can we think of another way to display the error message? For now maybe with a error toast? You can use:

```tsx
toaster.create({
              description,
            })
```